tools/ensure_codegen_up_to_date.py

```diff
@@ -0,0 +1,70 @@
+"""
+ensure_codegen_up_to_date
+-------------------------
+Ensure changes to code generation are committed to repository
+"""
+
+import glob
+import os
+import subprocess
+import sys
+
+
+def configure_git_credentials():
+
+    """Configures git user name and email with dummy name and email"""
+
+    if os.system('git config user.email "dummy@ni.com"') != 0:
+        sys.exit("Error: Unable to configure \"user email\" using git")
+
+    if os.system('git config user.name "dummy"') != 0:
+        sys.exit("Error: Unable to configure \"user name\" using git")
+
+
+def clean_codegen_files():
+
+    """Before code generation clean the existing codegen files"""
+
+    if os.system("tox -e clean") != 0:
+        sys.exit("Error: Unable to clean the repo using \"tox -e clean\"")
+
+
+def create_codegen_files():
+
+    """create codegen files"""
+
+    if os.system("tox -e codegen") != 0:
+        sys.exit("Error: Unable to generate code using \"tox -e codegen\"")
+
+
+def convert_all_files_to_unix_line_endings():
+
+    """Changes the line endings of all the files in the repo to LF"""
+
+    nimi_python_files = glob.glob(os.path.join(os.getcwd(), '**'), recursive=True)
+    for file_name in nimi_python_files:
+        subprocess.call(['dos2unix', file_name], shell=False)
+
+
+def check_no_dirty_files():
+
+    '''Checks if there are any modified files'''
+
+    if int(subprocess.check_output("git status -s -uno | wc -l", shell=True).decode().strip("b'\\n'")) != 0:
+        list_of_changed_files = subprocess.check_output("git status -s -uno", shell=True).decode()
+        file_diffs = subprocess.check_output("git diff", shell=True).decode()
+        sys.exit(f"The following code generated files do not match what is commited to Git. Run codegen and include the generated files in the PR.\n{list_of_changed_files}\n File diffs: \n{file_diffs} ")
+    print("All changes to code generation are committed to repository")
+
+
+if __name__ == "__main__":
+    """
+    Main
+    """
+    configure_git_credentials()
+
+    clean_codegen_files()
+    create_codegen_files()
+
+    convert_all_files_to_unix_line_endings()
```

Comment:
Can you explain why this is needed?
I suspect the correct fix to this issue is in `git config`.
Simply cloning the repo and running codegen should not leave you with files different than what is committed but that appears to be what you are correcting for in this function.

Reply:
When codegen runs in the TravisCI build, nidigital/session.py's line endings change. For some reason, it only happens to that file. It's causing the CI to fail, so Shreyas suggested working around it by normalizing the line endings to LF and running the command on all the files for simplicity.

Reply:
Since the workaround was proving difficult, I went ahead and root-caused the issue and put up a fix: https://github.com/ni/nimi-python/pull/1728

With that, you don't need the workaround.

Reply:
TravisCI is now passing